frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx

```diff
@@ -1,44 +1,150 @@
+
+  it('when call verify function given a invalid token then should got email and token fields error message', async () => {
+    server.use(
+      rest.post(MOCK_BOARD_URL_FOR_JIRA, (_, res, ctx) => {
+        return res(ctx.status(HttpStatusCode.Unauthorized));
+      }),
+    );
+
+    const { result } = renderHook(() => useVerifyBoardEffect());
+    await act(() => {
+      result.current.verifyJira({
+        type: 'jira',
+        boardId: '1',
+        site: 'fake',
+        token: 'fake-token',
+        startTime: null,
+        endTime: null,
+      });
     });
+
+    await waitFor(() => {
+      const emailFiled = result.current.formFields.find((field) => field.name === 'email');
+      const tokenField = result.current.formFields.find((field) => field.name === 'token');
+
+      expect(emailFiled?.errorMessage).toBe('Email is incorrect !');
+      expect(tokenField?.errorMessage).toBe(
+        'Token is invalid, please change your token with correct access permission !',
+      );
+    });
+  });
+
+  it('when call verify function given a invalid site then should got site field error message', async () => {
+    server.use(
+      rest.post(MOCK_BOARD_URL_FOR_JIRA, (_, res, ctx) => {
+        return res(
+          ctx.status(HttpStatusCode.NotFound),
+          ctx.json({
+            hintInfo: 'site not found',
+          }),
+        );
+      }),
+    );
+
     const { result } = renderHook(() => useVerifyBoardEffect());
+    await act(() => {
+      result.current.verifyJira({
+        type: 'jira',
+        boardId: '1',
+        site: 'fake',
+        token: 'fake-token',
+        startTime: null,
+        endTime: null,
+      });
+    });
+
+    await waitFor(() => {
+      const site = result.current.formFields.find((field) => field.name === 'site');
 
-    act(() => {
-      result.current.verifyJira(MOCK_BOARD_VERIFY_REQUEST_PARAMS);
+      expect(site?.errorMessage).toBe('Site is incorrect !');
     });
+  });
 
-    expect(result.current.errorMessage).toEqual(
-      `${MOCK_BOARD_VERIFY_REQUEST_PARAMS.type} ${VERIFY_FAILED}: error message`,
+  it('when call verify function given a invalid board id then should got board id field error message', async () => {
+    server.use(
+      rest.post(MOCK_BOARD_URL_FOR_JIRA, (_, res, ctx) => {
+        return res(
+          ctx.status(HttpStatusCode.NotFound),
+          ctx.json({
+            hintInfo: 'boardId not found',
+          }),
+        );
+      }),
     );
+
+    const { result } = renderHook(() => useVerifyBoardEffect());
+    await act(() => {
+      result.current.verifyJira({
+        type: 'jira',
+        boardId: '1',
+        site: 'fake',
+        token: 'fake-token',
+        startTime: null,
+        endTime: null,
+      });
+    });
+
+    await waitFor(() => {
+      const boardId = result.current.formFields.find((field) => field.name === 'boardId');
+
+      expect(boardId?.errorMessage).toBe('Board Id is incorrect !');
```

Comment:
are u sure there is a blank between `!` and `incorrect`?

Reply:
yes, i'm sure